### DISCUSSION TITLE
Adding XMP to Webp

### DISCUSSION THREAD

**Discussion by jarodium** [ImageMagick/ImageMagick#3617] [Help]
Hello, 

I'm having some issues adding XMP metadata into a WebP file. I'm using Imagick in PHP 7.2 and using the latest imagemagick with webp support. On some files, more accurately JPG and PNG, i can just setImageProfile/getImageProfile to set/get XMP, but with webp it seems it's not working. You can check my code [here](https://stackoverflow.com/questions/67136507/imagick-setprofile-xmp-in-webp-file-format). I've also tried using setOption but the docs [here](https://imagemagick.org/script/escape.php) don't contain any information about adding XMP either. Only EXIF and IPTC are mentioned.

Can anyone shed some light on this matter? Is this a lack of support issue or its just that WebP simply does not allow for color profiles, in order to be kept low on size? I've tried adding XMP metadata with libvips and it works just fine, but i'm using Imagemagick since some systems I work with don't provide vips and I need a backup plan.

Thank you in advance
Best regards
jarodium

**Comment by dlemstra** [comment]
It turns out that you found a bug. We were reading and writing the XMP chunk as `XMP` instead of `XMP `. The correct chunk name includes a space at the end. This will be resolved in the next version of ImageMagick.

**Reply by jarodium** [reply]
Hello

I have now been able to write XMP to the file using **imageSetProfile('xmp',$xml)** in PHP.
Nonetheless, reading profiles with **getImageProfiles('xmp')** does not seem to read anything from 'xmp' "profile". I only get exif binary information in the dump.

Using [http://exif.regex.info/](http://exif.regex.info/) I am able to see a RIFF section with my XMP's data inside ( check below ).

![image](https://user-images.githubusercontent.com/1470446/121945062-42175a00-cd4b-11eb-899f-68a49e4a2b51.png)

Do I need to update PHP extension as well? Or shoud I be using another way to get the XMP?

**Comment by dlemstra** [comment]
Is it possible to share that file? You can zip it and attach it here. It looks like that file has the profile stored incorrectly. I might be able to add a patch for this situation.

**Reply by jarodium** [reply]
@dlemstra 

I have sent it by email, to the address in your profile.

Thank you.

Best regards
jarodium

**Reply by dlemstra** [reply]
Thanks for the image @jarodium and I just pushed a patch that will allow support for reading those incorrectly stored `XMP` profiles. This will become available in the next release.

**Reply by jarodium** [reply]
@dlemstra thank you.

I'll await for the new version to come out.
